### ISSUE TITLE
IO::Handle method say should ignore $\

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9175]
Migrated from [rt.perl.org#49266](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=49266) (status was 'resolved')

Searchable as RT49266$

**Comment by p5pRT** [comment]
### From zsbana@gmail.com

#### Created by ambrus@math.bme.hu



The say method of IO&#8203;::Handle behaves differently from the say
primitive if the ORS is set\.

Just like perlfunc says\, the say builtin ignores the $\\ variable\.  The
say method of IO&#8203;::Handle\, however\, outputs it after the newline\.

I believe the method should behave the same as the builtin\.

Here's an example&#8203;:

\[am\]king ~$ perl \-wE 'use IO&#8203;::Handle; $\\ = "i"; STDOUT\->say\("h"\);' | od \-c
0000000   h  \\n   i
0000003
\[am\]king ~$ perl \-wE 'use IO&#8203;::Handle; $\\ = "i"; say STDOUT "h";' | od \-c
0000000   h  \\n
0000002

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low

Site configuration information for perl 5.10.0:

Configured by ambrus at Tue Dec 18 19:35:54 CET 2007.

Summary of my perl5 (revision 5 version 10 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.6.20.6, archname=i686-linux
    uname='linux king 2.6.20.6 #3 smp tue may 29 09:42:52 cest 2007
i686 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-fno-strict-aliasing -pipe
-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion='', gccversion='4.1.1', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /usr/lib64
    libs=-lnsl -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.3.6.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.6'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib'

Locally applied patches:



@INC for perl 5.10.0:
    /usr/local/lib/perl5/5.10.0/i686-linux
    /usr/local/lib/perl5/5.10.0
    /usr/local/lib/perl5/site_perl/5.10.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.10.0
    /usr/local/lib/perl5/site_perl/5.8.8
    /usr/local/lib/perl5/site_perl
    .


Environment for perl 5.10.0:
    HOME=/home/ambrus
    LANG (unset)
    LANGUAGE=en_HU:en_US:en_GB:en
    LC_CTYPE=hu_HU
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/ambrus/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From ambrus@math.bme.hu

The following patch seems to make the example code work correctly\.  

<details><summary>Inline Patch</summary>

```diff
--- perl-5.10.0/ext/IO/lib/IO/Handle.pm	2008-01-08 11:01:43.000000000 +0100
+++ perl-5.10.0/ext/IO/lib/IO/Handle.pm	2008-01-08 11:03:19.000000000 +0100
@@ -411,9 +411,10 @@
 
 sub say {
     @_ or croak 'usage: $io->say(ARGS)';
     my $this = shift;
-    print $this @_, "\n";
+    local $\ = "\n";
+    print $this @_;
 }
 
 sub getline {
     @_ == 1 or croak 'usage: $io->getline()';
```
</details>

**Comment by p5pRT** [comment]
### From @rgs

Fixed in bleadperl by change \#33258\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Fixed in bleadperl by change \#33258\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'new' to 'resolved'
